main_test.go

```diff
@@ -0,0 +1,734 @@
+package tests
```

Comment:
This looks like a questionable practice. If this needs to be a separate package, then let's have a corresponding directory for this package.

Reply:
Also: strongly recommend moving this into its own PR. That will significantly reduce the size of this PR, and allow proper review of both this and the subsequent PR, focusing on the specific aspect they are targeting.